### ISSUE TITLE
Appium webdriver instantiation fails for iOS

### ISSUE THREAD

**Issue by dmshantanu** [appium/appium#9581]
## The problem

All I did was upgrade to appium 1.6.5 from appium 1.6.4. I started seeing "xcodebuild failed with code 65".  This was working completely fine with appium 1.6.4. Link to appium logs below.

https://gist.github.com/anonymous/183eebf480219d07e21957044a8d89e3

I uninstalled appium (npm uninstall -g appium) and installed again (npm install -g appium@1.6.5).
Still I see the same problem

**Comment by mykola-mokhnach** [comment]
Error 65 means WDA signing issue. Reinstalling/uninstalling Appium rewrites WDA sources, so you need to re-setup signing if you don't use an external .xcconfig. See https://github.com/appium/appium-xcuitest-driver/blob/master/docs/real-device-config.md for more details.

**Comment by dmshantanu** [comment]
@mykola-mokhnach  I had followed all those steps. I was also able to build  WebdriverAgentRunner and install it on iPhone using xcode. However, when I ran my tests using appium I get  the error "xcodebuild failed with code 65". Also I noticed that when I built WebdriverAgentRunner, xcode started  WebdriverAgentRunner and then it suddenly stopped as if it crashed. Xcode still says that build succeeded. I do not know what I am missing. Can you please help?

**Comment by mykola-mokhnach** [comment]
It might be that when you execute your automated via Jenkins then its agent does not have access to the keychain, where your signing key is stored. Try to play with _keychainPath_ and _keychainPassword_ capabilities in such case.

**Comment by dmshantanu** [comment]
@mykola-mokhnach  I am not using Jenkins.I am running it on my laptop. Xcode has access to my keychain and is able to use it to build the WebDriverAgentRunner.

Also when I switchback to appium 1.6.4 setup I dont see this problem.

**Comment by mykola-mokhnach** [comment]
Actually, version 1.6.5 is quite and old one. The recent stable one is 1.7.1
Try it and if it still does not work then attach the log with _showXcodeLog_ capability enabled.

**Comment by dmshantanu** [comment]
Will it work with Xcode 8.3.2 ? Thats what I am using.

**Comment by dmshantanu** [comment]
https://gist.github.com/dmshantanu/73ec19613e02b29ccf151b91939ea01f

I upgraded to appium 1.7.1 and I see the same problem. Please find logs attached with showXcodeLog set to true.

**Comment by imurchie** [comment]
I've run into this when upgrading because things get left over from the old version, and Apple can't handle it.

Try deleting all remnants of the build (from your derived data folder (`sudo rm -rf ~/Library/Developer/Xcode/DerivedData/WebDriverAgent-*`) and off the device.

**Comment by dmshantanu** [comment]
Thanks @imurchie . Will try it out. As of now I got it to work by copying the appium folder under node_modules from another working setup.

**Comment by dmshantanu** [comment]
Thanks @imurchie It worked. I could install appium 1.7.1

**Comment by dmshantanu** [comment]
@imurchie  @tfcardoso  I am ran into same issue on another machine.  I deleted  xcode remnants using "sudo rm -rf ~/Library/Developer/Xcode/DerivedData/WebDriverAgent-*" .

However, it did not help. I am running into same problem again. I uninstalled and reinstalled appium 1.7.1 multiple times. Is there anything else I can do?

I am using xcode version 8.3.2

Here is a gist for the command:
xcodebuild -project WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -destination 'id=7f569159bd171c6f7cf38c639de026316aa5bc3b' test
https://gist.github.com/dmshantanu/ec80b0a038ee3e5a95d40af1315b16dd
